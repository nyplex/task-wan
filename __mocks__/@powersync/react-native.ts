module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
