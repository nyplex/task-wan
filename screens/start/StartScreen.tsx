import ThemedButton from "@/components/ThemedButton";
import ThemedText from "@/components/ThemedText";
import { Box } from "@/components/ui/box";
import { Image } from "expo-image";
import { useRouter } from "expo-router";
import { useState } from "react";
import { Slider } from "./components/Slider";
import SliderIndicator from "./components/SliderIndicator";

const StartScreen = () => {
  const router = useRouter();
  const [index, setIndex] = useState(0);

  return (
    <Box className="flex-1 py-safe-offset-4 bg-backrgound">
      <Box className="mb-4 px-4">
        <SliderIndicator length={3} currentIndex={index} />
      </Box>
      <Slider onIndexChange={(i) => setIndex(i)}>
        <Box className="flex-1 px-4">
          <Box>
            <Image
              source={require("@/assets/images/start-1.png")}
              contentFit="contain"
              style={{ width: "100%", height: 400 }}
            />
          </Box>
          <Box className="flex-1 w-full h-full px-4">
            <ThemedText size="body" weight="semi-bold" className="text-center">
              Easy Time Management
            </ThemedText>
            <ThemedText size="bodyS" className="text-center mt-2">
              With management based on priority and daily tasks, it will give
              you convenience in managing and determining the tasks that must be
              done first
            </ThemedText>
          </Box>
        </Box>
        <Box className="flex-1 px-4">
          <Box>
            <Image
              source={require("@/assets/images/start-2.png")}
              contentFit="contain"
              style={{ width: "100%", height: 400 }}
            />
          </Box>
          <Box className="flex-1 w-full h-full px-4">
            <ThemedText size="body" weight="semi-bold" className="text-center">
              Increase Work Effectiveness
            </ThemedText>
            <ThemedText size="bodyS" className="text-center mt-2">
              Time management and the determination of more important tasks will
              give your job statistics better and always improve
            </ThemedText>
          </Box>
        </Box>
        <Box className="flex-1 px-4">
          <Box>
            <Image
              source={require("@/assets/images/start-3.png")}
              contentFit="contain"
              style={{ width: "100%", height: 400 }}
            />
          </Box>
          <Box className="flex-1 w-full h-full px-4">
            <ThemedText size="body" weight="semi-bold" className="text-center">
              Reminder Notification
            </ThemedText>
            <ThemedText size="bodyS" className="text-center mt-2">
              The advantage of this application is that it also provides
              reminders for you so you don't forget to keep doing your
              assignments well and according to the time you have set
            </ThemedText>
          </Box>
        </Box>
      </Slider>

      <Box className="px-4">
        <ThemedButton
          title="Get Started"
          onPress={() => router.navigate("/(app)/Login")}
        />
      </Box>
    </Box>
  );
};

export default StartScreen;
