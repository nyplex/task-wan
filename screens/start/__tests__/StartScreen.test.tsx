import { render, fireEvent } from "@testing-library/react-native";
import StartScreen from "../StartScreen";

// Mock all child components to isolate StartScreen
const Slider = ({ children }: any) => <>{children}</>;
Slider.displayName = "Slider";
const SliderIndicator = () => <></>;
SliderIndicator.displayName = "SliderIndicator";
const SliderChild = () => <></>;
SliderChild.displayName = "SliderChild";

jest.mock("../components/Slider", () => Slider);
jest.mock("../components/SliderIndicator", () => SliderIndicator);
jest.mock("../components/SliderChild", () => SliderChild);
jest.mock("expo-router", () => ({ useRouter: () => ({ navigate: jest.fn() }) }));

describe("StartScreen", () => {
  it("renders without crashing", () => {
    render(<StartScreen />);
  });

  it("renders Box, Slider, SliderIndicator, SliderChild, and Button", () => {
    const { UNSAFE_queryAllByType } = render(<StartScreen />);
    expect(UNSAFE_queryAllByType(Slider).length).toBe(1);
    expect(UNSAFE_queryAllByType(SliderIndicator).length).toBe(1);
    expect(UNSAFE_queryAllByType(SliderChild).length).toBe(3);
  });

  it("calls router.navigate when Get Started is pressed", () => {
    const mockNavigate = jest.fn();
    jest.spyOn(require("expo-router"), "useRouter").mockReturnValue({ navigate: mockNavigate });
    const { getByText } = render(<StartScreen />);
    fireEvent.press(getByText("Get Started"));
    expect(mockNavigate).toHaveBeenCalledWith("/(app)/Login");
  });

  it("matches the rendered output snapshot", () => {
    const { toJSON } = render(<StartScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
