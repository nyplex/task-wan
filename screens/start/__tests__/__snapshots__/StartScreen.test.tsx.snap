// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen matches the rendered output snapshot 1`] = `
<View
  className="flex-1 py-safe-offset-4 bg-backrgound"
>
  <View
    className="mb-4 px-4"
  />
  <View
    className="px-4"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      className="group/button flex-row items-center justify-center data-[focus-visible=true]:web:outline-none data-[focus-visible=true]:web:ring-2 data-[disabled=true]:opacity-40 gap-2 data-[hover=true]:bg-primary-600 border-primary-300 data-[hover=true]:border-primary-400 data-[active=true]:border-primary-500 data-[focus-visible=true]:web:ring-indicator-info px-5 based-styles w-full h-[48px] rounded-[10px] bg-primary-0 data-[active=true]:bg-primary-50"
      collapsable={false}
      dataSet={
        {
          "active": "false",
          "disabled": "false",
          "focus": "false",
          "focusVisible": "false",
          "hover": "false",
        }
      }
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      states={
        {
          "active": false,
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
        }
      }
      testID="Button"
    >
      <View
        className="flex-row items-center justify-center gap-2 flex-1"
      >
        <Text
          className="based-styles font-regular text-[16px] tracking-[0px] text-white"
        >
          Get Started
        </Text>
      </View>
    </View>
  </View>
</View>
`;
